### ISSUE TITLE
Question on "Waited time".

### ISSUE THREAD

**Issue by cbmeeks** [glowroot/glowroot#631]
I'm new to Glowroot and I've just installed it on a legacy application we have (some of it written in 2008).

At first glance I found a very significant issue in our app that is causing major problems.  It appears some method has gone wild and never quits.

Anyway, Glowroot is displaying the following information:

**JVM Thread Stats**
CPU time: 26,390.6 milliseconds
Blocked time: 5,062.0 milliseconds
Waited time: 2,658,256.0 milliseconds
Allocated memory: 2.3 GB 

As you can see, that Waited time is astronomical.  I've been watching that slow trace for a while and it just keeps climbing.

My question is, how can I identify what Glowroot is saying the "Waited time" actually is?
I see hundreds of queries that probably took the 26K CPU time, but what is the "Waited time"?

I've searched for documentation but cannot find it.

Thanks.

**Comment by trask** [comment]
Hi @cbmeeks, wait time could be time spent waiting for response from database, check the thread profile for that trace to see where in the code it's spending time.

**Comment by ernestoalejandrocu** [comment]
hi @trask , can you  help me with the "blocked time" indicator, what it is? whats mean?, thanks

**Comment by Never-lab** [comment]
Already answered by @trask.

**Waited time** (and **Blocked time**) come from Java thread MXBean stats on the transaction thread:
- **Waited** ≈ time in `WAITING` / `TIMED_WAITING` (e.g. `Object.wait`, `LockSupport.park`, many blocking waits — often including waiting on a DB/socket response depending on how the driver blocks).
- **Blocked** ≈ time waiting to enter a monitor (`BLOCKED`).

Huge waited time with modest CPU usually means the thread is parked/blocked most of the wall clock — use that trace’s **thread profile** (and entries/queries) to see *where*, not only the JVM Thread Stats totals.

Same for the follow-up on blocked time. Docs Qs → [Discussions](https://github.com/glowroot/glowroot/discussions).

OK to close as support.

@nowheresly
